### ISSUE TITLE
Entity creation from api don't work as expected

### ISSUE THREAD

**Issue by tekmans** [glpi-project/glpi#18476]
### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

10.0.16

### Bug description

I have scripted the installation of glpi and glpi plugin, next i had rename root entity and create different entities

Those son entities are wrongly created as

![Image](https://github.com/user-attachments/assets/0d9ff43f-8f25-4593-984a-0e7d7951a667)

As we can see son entities are not clickable

On the main screen, i can see only one entity
![Image](https://github.com/user-attachments/assets/6f548f5a-5541-4769-a6a1-b446b7705afa)

And it becomes impossible to create rule based on entity (entities are missing)
![Image](https://github.com/user-attachments/assets/81622441-553e-4ad6-9fb1-21fa4ca5ed81)

If on my script i bypass this
```
#- name: GLPI - Enable glpi plugin after db init
#  ansible.builtin.command: php /usr/share/glpi/bin/console {{ item }} glpiinventory
#  with_items:
#    - 'glpi:plugin:install --username=root'
#    - 'glpi:plugin:activate'
```
And do it from glpi interface everything works as a charme
![Image](https://github.com/user-attachments/assets/1384e118-7d5e-42a7-90b3-c099d2586068)
![Image](https://github.com/user-attachments/assets/9ebedb13-9f62-4017-8c2b-aa73e744f9d9)

And rules are possible
![Image](https://github.com/user-attachments/assets/85dc151e-e819-42f5-8cf4-5f6d6573929f)

Why the plugin is working differently if is activated by command line ?

### Relevant log output

```shell

```

**Comment by trasher** [comment]
Sounds like an issue on your "script", but since you do not provide any information, can't say.

Sounds like a support request for now, not a bug.

**Comment by tekmans** [comment]
Here is the script (install glpi, create first entity before glpiinventory installation and activation)
![Image](https://github.com/user-attachments/assets/45395a61-0f6f-41d8-b3a5-0f5177b328cd)


The result
![Image](https://github.com/user-attachments/assets/b2d6b66a-74b9-44c6-bb9d-ec0407f692ff)

Here the manually glpiinventory installation

![Image](https://github.com/user-attachments/assets/8b5bfc98-a4cc-4ae6-9a32-a5ba0d12a02e)

The result
- demo is correctly created (we see the link)
- entite1 is not correctly created (no link)

![Image](https://github.com/user-attachments/assets/817dd818-08a7-47bd-b0f7-394b18b758d1)

Rules on entities
And here we see also, demo is present and entite1 no
![Image](https://github.com/user-attachments/assets/a8d2d569-4d1c-4ae1-8779-d6b3c5fbb41e)

I hope it's more clear

**Comment by cedric-anne** [comment]
Hi,

Maybe this is related to a caching issue.

Here is the possible workarounds:
 - executing the `bin/console cache:clear` command after the entity creation may fix the issue if it is related to a general caching issue;
 - log out then log in may fix the issue if it is related to a caching issue at the session level.

Also, please try to reproduce on GLPI 10.0.17. We recently fixed issues on the entity tree caching mechanism, maybe you are facing one of them.

**Comment by trasher** [comment]
Cache in database may also cause issues; see https://github.com/glpi-project/glpi/issues/18023#issuecomment-2423122549

**Comment by tekmans** [comment]
Well done !! clear cache works

Because i tried log off, log in, spent a lot off time searching relation between script, version of glpi, plugin and so one....

I have double check on very old installation, i clear the cache and it's work also.

@cedric-anne i tried 10.0.17, it works out of the box :)

**Comment by tekmans** [comment]
@cedric-anne I have tried again with 10.0.17, and failed again... i need to clear cache
Is it these function of clearing cache available through api ?

If you need extra log or something, please ask

**Comment by cedric-anne** [comment]
> Is it these function of clearing cache available through api ?

No, and they should not. The cache is expected to be automatically cleaned or refreshed if any action that make it stale is executed.

Do you have a unique server, or do you have multiple servers ? If the cache is not correctly refreshed, either there is a bug, either there the CLI and the API calls are made on contexts that are not sharing the cache storage.
